dcos/marathon.py

```diff
@@ -741,10 +743,30 @@ def remove_pod(self, pod_id, force=False):
         :rtype: None
         """
 
-        path = 'v2/pods' + _normalize_marathon_id_path(pod_id)
+        path = self._marathon_id_path_join('v2/pods', pod_id)
         params = {'force': 'true'} if force else None
         self._rpc.http_req(http.delete, path, params=params)
 
+    @staticmethod
+    def _marathon_id_path_join(url_path, id_path):
+        """Concatenates a URL path with a Marathon "ID path", ensuring the
+        result is well-formed.
+
+        The path and the ID will be joined with a single forward slash (/),
+        all trailing slashes in the ID will be removed, and the ID will have
+        all URL-unsafe characters escaped, as if by urllib.parse.quote().
+
+        :param url_path: the path portion of a URL
+        :type url_path: str
+        :param path_id: a Marathon "ID path", e.g. app ID or group ID
+        :type path_id: str
+        :returns: the path with the ID appended
+        :rtype: str
+        """
+
+        normalized_id_path = urllib.parse.quote(id_path.strip('/'))
+        return posixpath.join(url_path, normalized_id_path)
```

Comment:
Based on the note, `os.path` works for all platforms, and `posixpath` only for unix:
```
The os.path module is always the path module suitable for the operating system Python is running on, and therefore usable for local paths. However, you can also import and use the individual modules if you want to manipulate a path that is always in one of the different formats. They all have the same interface:
posixpath for UNIX-style paths
ntpath for Windows paths
macpath for old-style MacOS paths
```